src/EditorFeatures/Core/CommandHandlers/GoToNextAndPreviousMethodCommandHandler.cs

```diff
@@ -0,0 +1,185 @@
+
+using System;
+using System.ComponentModel.Composition;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.CodeAnalysis.Editor.Commands;
+using Microsoft.CodeAnalysis.Editor.Host;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.SuggestionSupport;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Outlining;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.CommandHandlers
+{
+    [ExportCommandHandler(PredefinedCommandHandlerNames.GoToNextAndPreviousMethod, ContentTypeNames.RoslynContentType)]
+    internal class GoToNextAndPreviousMethodCommandHandler : ICommandHandler<GoToNextMethodCommandArgs>, ICommandHandler<GoToPreviousMethodCommandArgs>
+    {
+        private readonly IWaitIndicator _waitIndicator;
+        private readonly IOutliningManagerService _outliningManagerService;
+
+        [ImportingConstructor]
+        public GoToNextAndPreviousMethodCommandHandler(IWaitIndicator waitIndicator, IOutliningManagerService outliningManagerService)
+        {
+            _waitIndicator = waitIndicator;
+            _outliningManagerService = outliningManagerService;
+        }
+
+        public CommandState GetCommandState(GoToNextMethodCommandArgs args, Func<CommandState> nextHandler)
+            => GetCommandState(args.SubjectBuffer, args.TextView, nextHandler);
+
+        public void ExecuteCommand(GoToNextMethodCommandArgs args, Action nextHandler)
+            => ExecuteCommand(nextHandler, args.SubjectBuffer, args.TextView, next: true);
+
+        public CommandState GetCommandState(GoToPreviousMethodCommandArgs args, Func<CommandState> nextHandler)
+            => GetCommandState(args.SubjectBuffer, args.TextView, nextHandler);
+
+        public void ExecuteCommand(GoToPreviousMethodCommandArgs args, Action nextHandler)
+            => ExecuteCommand(nextHandler, args.SubjectBuffer, args.TextView, next: false);
+
+        private static CommandState GetCommandState(ITextBuffer subjectBuffer, ITextView textView, Func<CommandState> nextHandler)
+        {
+            var document = subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            var caretPoint = textView.GetCaretPoint(subjectBuffer);
+            return IsAvailable(document, caretPoint) ? CommandState.Available : nextHandler();
+        }
+
+        private static bool IsAvailable(Document document, SnapshotPoint? caretPoint)
+        {
+            var documentSupportsSuggestionService = document.Project.Solution.Workspace.Services.GetService<IDocumentSupportsSuggestionService>();
+            if (document == null || !document.SupportsSyntaxTree || !documentSupportsSuggestionService.SupportsNavigationToAnyPosition(document))
```

Comment:
You're checking if the Document is null *after* the line where you ```<dot>``` off of it to get the project.